### PR TITLE
evm vanchor leaves watcher #160

### PR DESCRIPTION
**Pull request by salman01zp**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by drewstone** [comment]
Need to fix CI errors before merge. Will review soon. Try merging master if this hasn't happened.

File: src/service.rs [hunk 1]
```diff
@@ -374,6 +385,64 @@ fn start_dkg_pallet_watcher(
...
 
+/// Starts the event watcher for EVM VAnchor events.
+///
+/// Returns Ok(()) if successful, or an error if not.
+///
+/// # Arguments
+///
+/// * `ctx` - RelayContext reference that holds the configuration
+/// * `config` - VAnchor contract configuration
+/// * `client` - DKG client
```
Comment (shekohex) [inline src/service.rs:396]:
```suggestion
/// * `client` - EVM Chain api client
```

File: tests/lib/localTestnet.ts [hunk 2]
```diff
@@ -215,49 +330,126 @@ export class LocalChain {
...
-              chain: (await anchor.contract.getChainId()).toString(),
-              address: anchor.getAddress(),
-            }))
-          ),
+      contracts: contracts,
+    };
+    return chainInfo;
+  }
+
+  public async getVAnchorChainConfig(
```
Comment (shekohex) [inline tests/lib/localTestnet.ts:370]: make it private?
```suggestion
  private async getVAnchorChainConfig(
```

File: tests/lib/localTestnet.ts [hunk 3]
```diff
@@ -194,7 +309,7 @@ export class LocalChain {
     return val;
   }
 
-  public async exportConfig(
+  public async getAnchorChainConfig(
```
Comment (shekohex) [inline tests/lib/localTestnet.ts:312]: shall be private

```suggestion
  private async getAnchorChainConfig(
```

**Review by shekohex — CHANGES_REQUESTED** [review]
LGTM!! Good job!

A few changes and CI shall pass, and we can merge this. but if the CI starts to act randomly again we can merge in `master` for now.

**Review by shekohex — APPROVED** [review]

**Pull request merged**